### PR TITLE
[SR SQL Planner Summer Camp] add ShowRepositories analyzer

### PR DESCRIPTION
**Pull request by melt-code**
## What type of PR is this：
- [ ] bug
- [x] feature
- [ ] enhancement
- [ ] refactor
- [ ] others

## Which issues of this PR fixes ：

Fixes #7951

## Problem Summary(Required) ：

add ShowRepositories analyzer

### CODE REVIEW

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 14 / 17 (82.35%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/ast/AstVisitor.java|0|1|00.00%|[418]|
|:large_blue_circle:|com/starrocks/analysis/ShowRepositoriesStmt.java|1|2|50.00%|[52]|
|:large_blue_circle:|com/starrocks/sql/analyzer/ShowRepositoriesAnalyzer.java|10|11|90.91%|[11]|
|:large_blue_circle:|com/starrocks/sql/analyzer/Analyzer.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/parser/AstBuilder.java|1|1|100.00%|[]|

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 11 / 13 (84.62%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/ast/AstVisitor.java|0|1|00.00%|[443]|
|:large_blue_circle:|com/starrocks/sql/analyzer/ShowRepositoriesAnalyzer.java|6|7|85.71%|[9]|
|:large_blue_circle:|com/starrocks/sql/analyzer/Analyzer.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/parser/AstBuilder.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/analysis/ShowRepositoriesStmt.java|2|2|100.00%|[]|

File: fe/fe-core/src/main/java/com/starrocks/sql/analyzer/ShowRepositoriesAnalyzer.java [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+public class ShowRepositoriesAnalyzer {
+
+    public static void analyze(ShowRepositoriesStmt showRepositoriesStmt, ConnectContext session) {
+        new ShowRepositoriesAnalyzerVisitor().analyze(showRepositoriesStmt, session);
+    }
+
+    public static class ShowRepositoriesAnalyzerVisitor extends AstVisitor<Void, ConnectContext> {
+        public void analyze(ShowStmt statement, ConnectContext session) {
+            visit(statement, session);
+
```
Comment (Youngwb) [inline fe/fe-core/src/main/java/com/starrocks/sql/analyzer/ShowRepositoriesAnalyzer.java:18]:
```suggestion
```

File: fe/fe-core/src/main/java/com/starrocks/sql/ast/AstVisitor.java [hunk 2]
```diff
@@ -438,6 +439,10 @@ public R visitShowHistogramStatsMetaStatement(ShowHistogramStatsMetaStmt stateme
         return visitShowStatement(statement, context);
     }
 
+    public R visitShowRepositoriesStmt(ShowRepositoriesStmt statement, C context) {
+        return visitStatement(statement, context);
```
Comment (Youngwb) [inline fe/fe-core/src/main/java/com/starrocks/sql/ast/AstVisitor.java:443]:
```suggestion
        return visitShowStatement(statement, context);
```

File: fe/fe-core/src/main/java/com/starrocks/sql/ast/AstVisitor.java [hunk 3]
```diff
@@ -438,6 +439,10 @@ public R visitShowHistogramStatsMetaStatement(ShowHistogramStatsMetaStmt stateme
         return visitShowStatement(statement, context);
     }
 
+    public R visitShowRepositoriesStmt(ShowRepositoriesStmt statement, C context) {
```
Comment (Youngwb) [inline fe/fe-core/src/main/java/com/starrocks/sql/ast/AstVisitor.java:442]: plz put it in the right place

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 11 / 13 (84.62%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/ast/AstVisitor.java|0|1|00.00%|[422]|
|:large_blue_circle:|com/starrocks/sql/analyzer/ShowRepositoriesAnalyzer.java|6|7|85.71%|[9]|
|:large_blue_circle:|com/starrocks/sql/analyzer/Analyzer.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/parser/AstBuilder.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/analysis/ShowRepositoriesStmt.java|2|2|100.00%|[]|

File: fe/fe-core/src/main/java/com/starrocks/sql/analyzer/ShowRepositoriesAnalyzer.java (hunk 1 above)
Comment (melt-code) [reply fe/fe-core/src/main/java/com/starrocks/sql/analyzer/ShowRepositoriesAnalyzer.java]: It has been modified

File: fe/fe-core/src/main/java/com/starrocks/sql/ast/AstVisitor.java (hunk 2 above)
Comment (melt-code) [reply fe/fe-core/src/main/java/com/starrocks/sql/ast/AstVisitor.java]: It has been modified

File: fe/fe-core/src/main/java/com/starrocks/sql/ast/AstVisitor.java (hunk 3 above)
Comment (melt-code) [reply fe/fe-core/src/main/java/com/starrocks/sql/ast/AstVisitor.java]: It has been modified

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 11 / 13 (84.62%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/ast/AstVisitor.java|0|1|00.00%|[480]|
|:large_blue_circle:|com/starrocks/sql/analyzer/ShowRepositoriesAnalyzer.java|6|7|85.71%|[9]|
|:large_blue_circle:|com/starrocks/sql/analyzer/Analyzer.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/parser/AstBuilder.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/analysis/ShowRepositoriesStmt.java|2|2|100.00%|[]|

**Pull request closed without merging**